plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1071,7 +1074,7 @@ void producer_plugin::set_program_options(
          ("snapshots-dir", bpo::value<std::filesystem::path>()->default_value("snapshots"),
           "the location of the snapshots directory (absolute path or relative to application data dir)")
          ("read-only-threads", bpo::value<uint32_t>(),
-          "Number of worker threads in read-only execution thread pool. Max 8.")
+         ("Number of worker threads in read-only execution thread pool. Defaults to 0 if configured as producer, otherwise defaults to "s + std::to_string(producer_plugin_impl::_ro_default_threads_nonproducer) + ". Max "s + std::to_string(producer_plugin_impl::_ro_max_threads_allowed) + "."s).c_str())
```

Comment:
> `.c_str()`

`option_description` makes a copy of the `char*` passed in so this should be okay; not sure what a better pattern may be

Reply:
If `option_description` did not makes a copy of the char* passed, you could return `s.c_str()`, `s` being a `static std::string` declared in a lambda called immediately, but probably fine as is.